Source/Diagnostics/MultiDiagnostics.cpp

```diff
@@ -69,10 +69,18 @@ MultiDiagnostics::ReadParameters ()
 }
 
 void
-MultiDiagnostics::FilterComputePackFlush (int step, bool force_flush)
+MultiDiagnostics::FilterComputePackFlush (int step, bool force_flush, bool BackTransform)
 {
+    int i = 0;
     for (auto& diag : alldiags){
-        diag->FilterComputePackFlush (step, force_flush);
+        if (BackTransform == true) {
+            if (diags_types[i] == DiagTypes::BackTransformed)
```

Comment:
This makes me nervous: did `alldiags` also contains the reduced diags?

Things to check:
- let's access `diags_types` with `.at()` instead of `[]`